src/common/transformations/src/transformations/op_conversions/convert_batch_to_space.cpp

```diff
@@ -130,90 +121,117 @@ void ov::pass::ConvertBatchToSpace::convert_batch_to_space_by_elements() {
-
-        const std::vector<int64_t>& block_values = block_const->cast_vector<int64_t>();
-        const std::vector<int64_t>& crops_end_values = crops_end_const->cast_vector<int64_t>();
-
-        std::vector<int64_t> dispersed_shape(1);
-        dispersed_shape.insert(dispersed_shape.end(), data_shape.begin(), data_shape.end());
-        std::vector<size_t> axes_order(block_values.size() + 1);
-        std::vector<int64_t> squeezed_shape(data_shape.begin(), data_shape.end());
-        if (squeezed_shape.size() > block_values.size()) {
-            return false;
-        }
+        const auto shape_of_data = std::make_shared<opset3::ShapeOf>(data);
 
-        NodeVector new_ops;
+        const auto block = batch_to_space->input_value(1);
+        const auto crops_begin = batch_to_space->input_value(2);
+        const auto crops_end = batch_to_space->input_value(3);
+
+        // static data shape rank implies static block shape
+        const auto block_lenght = static_cast<int64_t>(block.get_shape()[0]);
+
+        const auto zero = opset3::Constant::create(element::i64, Shape{1}, {0});
+        const auto one = opset3::Constant::create(element::i64, Shape{1}, {1});
+        const auto two = opset3::Constant::create(element::i64, Shape{1}, {2});
 
+        std::shared_ptr<Node> dispersed_shape = std::make_shared<opset3::Concat>(OutputVector{one, shape_of_data}, 0);
+        const auto dispersed_shape_end =
+            opset3::Constant::create(element::i64, Shape{1}, {dispersed_shape->get_shape()[0]});
+
+        std::vector<size_t> axes_order(block_lenght + 1);
+        std::shared_ptr<Node> squeezed_shape = shape_of_data;
+        const auto squeezed_shape_end =
+            opset3::Constant::create(element::i64, Shape{1}, {squeezed_shape->get_shape()[0]});
+
+        NodeVector new_ops;
         std::shared_ptr<Node> flat_node = data.get_node_shared_ptr();
-        for (size_t block_idx = 1; block_idx < block_values.size(); ++block_idx) {
-            dispersed_shape[0] = block_values[block_idx];
-            dispersed_shape[1] /= block_values[block_idx];
-            const auto out_pattern_1 =
-                opset3::Constant::create(element::i64, Shape{dispersed_shape.size()}, dispersed_shape);
-            const bool special_zero = false;
-            flat_node = std::make_shared<ov::opset3::Reshape>(flat_node, out_pattern_1, special_zero);
+
+        std::shared_ptr<Node> dispersed_shape_carry_over;  // only assigned during first iteration
+        for (size_t block_idx = 1; block_idx < block_lenght; ++block_idx) {
+            const auto bidx = opset3::Constant::create(element::i64, Shape{1}, {block_idx});
+            const auto block_value = std::make_shared<opset8::Gather>(block, bidx, zero);
+
+            // dispersed_shape[0] = block[block_idx];
+            // dispersed_shape[1] /= block[block_idx];
+
+            const auto divided =
```

Comment:
```
   using namespace opset10;
   Output<Node> div, dispersed_shape;
   for (...)        
            const auto bidx = Constant::create(element::i64, Shape{1}, {block_idx});
            const auto current_block_value = std::make_shared<Gather>(block, bidx, zero);
            const auto block_value_1st_dim = std::make_shared<Gather>(dispersed_shape, one, zero);
            if (!div) {
                  div = make_shared<Divide>(block_value_1st_dim, current_block_value);
             } else {
                  div = make_shared<Divide>(div, current_block_value);
             }
            const auto dispersed_shape_tail = make_shared<Slice>(dispersed_shape, two, INT_MAX, one);
            dispersed_shape = make_shared<Concat>(OutputVector{block_value, div, dispersed_shape_tail}, 0);
            constexpr auto special_zero = false;
            flat_node = std::make_shared<Reshape>(flat_node, dispersed_shape, special_zero);
            // axis order calculation...

            const auto sq_shape_bidx =  make_shared<Gather>(squeezed_shape, bidx, zero);
            const auto sq_mul = make_shared<Multiply>(sq_shape_bidx , block_value);
            const auto sq_slice = std::make_shared<Slice>(squeezed_shape, one, block_idx + 1, one);
            const auto sq_shape_tail = make_shared<Slice>(squeezed_shape,
                                                Constant::create(element::i64, Shape{1}, {block_idx + 1}),
                                                INT_MAX,
                                                one);
            squeezed_shape = make_shared<Concat>(OutputVector{div, sq_slice, sq_multiplied, sq_shape_tail}, 0);                                    
            flat_node = std::make_shared<Reshape>(flat_node, squeezed_shape, special_zero);
            
            auto dispersed_shape_beg = make_shared<Slice>(dispersed_shape, 0, block_idx + 1, one);
            auto dispersed_shape_tail = make_shared<Slice>(dispersed_shape, block_idx + 2, INT_MAX, one)
            dispersed_shape = make_shared<Concat>(dispersed_shape_beg, sq_mul, dispersed_shape_tail);
```
let's try something like that, maybe it will help
please create necessary constants for INT_MAX,  block_idx + 2,  block_idx + 1